packages/nodejs/src/net/NodeJsHttpEndpoint.ts

```diff
@@ -0,0 +1,358 @@
+
+        if ("on" in optionsOrServer) {
+            ({ close, ready, server, notFound } = this.#bindToServer(optionsOrServer));
+        } else {
+            ({ close, ready, server, notFound } = this.#createDedicatedServer(optionsOrServer));
+        }
+
+        this.#server = server;
+        this.#ready = ready;
+        this.close = close;
+        this.#notFound = notFound;
+    }
+
+    get server() {
+        return this.#server;
+    }
+
+    #bindToServer(server: Server) {
+        return {
+            server,
+            ready: Promise.resolve(),
+            close: async () => {
+                this.http = undefined;
+                this.ws = undefined;
+            },
+            notFound: () => undefined,
+        };
+    }
+
+    #createDedicatedServer(options: NodeJsHttpEndpoint.Options) {
+        const server = createServer({ keepAlive: true });
+
+        const opts = {} as ListenOptions;
+
+        const address = AppAddress.for(options.address);
+        const { transport } = address;
+        switch (transport.kind) {
+            case "ip":
+                if (!address.isWildcardHost) {
+                    opts.host = address.host;
+                }
+                if (!address.isWildcardPort) {
+                    opts.port = address.portNum;
+                }
+                break;
+
+            case "unix":
+                const path = decodeURIComponent(address.hostname);
+                if (options.basePathForUnixSockets) {
+                    opts.path = resolve(options.basePathForUnixSockets, normalize(path));
+                } else {
+                    opts.path = normalize(path);
+                }
+                if (existsSync(opts.path)) {
+                    if (statSync(opts.path).isSocket()) {
+                        try {
+                            rmSync(opts.path);
+                        } catch (e) {
+                            throw new NetworkError(
+                                `Error deleting previous socket at ${opts.path}: ${asError(e).message}`,
+                            );
+                        }
+                    } else {
+                        throw new NetworkError(`UNIX socket path ${opts.path} exists and is not a socket`);
+                    }
+                }
+                break;
+
+            default:
+                throw new NetworkError(
+                    `Unsupported address type "${(options.address as any)?.type}" for HTTP endpoint`,
+                );
+        }
+
+        server.listen(opts);
+
+        return {
+            server,
+
+            ready: new Promise<void>((resolve, reject) => {
+                let settled = false;
+                server.once("listening", () => {
+                    if (settled) {
+                        return;
+                    }
+
+                    settled = true;
+                    resolve();
+                });
+                server.once("error", error => {
```

Comment:
should we have another error handle for after that initial one because else nodejs might throw uncatcheable exception. Also in general having an error handler on the server (at least if we have created that) is good, also for WS
But fine with adding later

Reply:
Yeah I guess we could, I'm not sure when node would emit such an error event so crashing didn't seem entirely inappropriate.  Difficult to know how to deal with an error with zero context...  But maybe better to just log, will need to deal with aftermath regardless

Reply:
I think logging for now is fine and then we need to see the real cases

Reply:
I added code to log a warning with the error message in these cases.  It sounds like Node may emit connection errors from failed connection attempts using this status code.  Which is different from "clientError" which I'm guessing is for errors from established connections.